src/Take.Elephant/IQueue.cs

```diff
@@ -39,6 +39,18 @@ public interface ISenderQueue<T>
         Task EnqueueAsync(T item, CancellationToken cancellationToken = default);
     }
 
+    public interface IStreamSenderQueue<T>: ISenderQueue<T>
```

Comment:
The name StreamQueue doesn't make much sense. One thing is a stream, another thing is a queue.

I suggest renaming to something like `IEventStreamPublisher` and `IEventStreamConsumer`, renaming the methods as well (Publish and Consume instead of Enqueue and Dequeue) and not inheriting from ISenderQuee interface.

Something like:

```csharp

public interface IEventStreamPublisher<TEvent, TKey>
{
   Task PublishAsync(TEvent @event, TKey key, CancellationToken cancellationToken)
}
public interface IEventStreamConsumer<TEvent, TKey>
{
   Task<(TEvent event, TKey key)> ConsumeAsync(CancellationToken cancellationToken)
}
```

Reply:
Thanks for the suggestions, I'm taking them to the new branch